### PR TITLE
Rename single file options

### LINKED ISSUE #12363: Single-file: Consider better names for options that extract files to disk
The following non-default optional settings for publishing single-file apps cause the host to extract certain files to disk at run-time.

[`IncludeNativeBinariesInSingleFile`
 `IncludeAllContentInSingleFile`
`IncludeSymbolsInSingleFile`](https://github.com/dotnet/designs/blob/master/accepted/2020/single-file/design.md#optional-settings)

These property names suggest the benefit (bundling more files) but not the cost (extraction).
Consider  renaming these options to suggest that content will be extracted.

This work is in lieu of, and probably better than  https://github.com/dotnet/sdk/issues/12095

@samsp-msft  can you please take a look into this? 

CC: @agocke  @vitek-karas

### PR DESCRIPTION
**Pull request by sbomer**
Fixes https://github.com/dotnet/sdk/issues/12363.

### CODE REVIEW

File: src/Tasks/Common/Resources/Strings.resx [hunk 1]
```diff
@@ -653,4 +653,12 @@ The following are names of parameters or literal values and should not be transl
     <value>NETSDK1137: It is no longer necessary to use the Microsoft.NET.Sdk.WindowsDesktop SDK. Microsoft.NET.Sdk can be used instead.</value>
     <comment>{StrBegin="NETSDK1137: "}</comment>
   </data>
+  <data name="CannotIncludeSymbolsInSingleFile" xml:space="preserve">
+    <value>NETSDK1138: Including symbols in a single file bundle is not supported when publishing for .NET5 or higher.</value>
+    <comment>{StrBegin="NETSDK1138: "}</comment>
+  </data>
+  <data name="CannotIncludeAllContentButNotNativeLibrariesInSingleFile" xml:space="preserve">
+    <value>NETSDK1139: Including all content in a single file bundle also includes native libraries. Either set IncludeNativeLibrariesForSelfExtract to true, or set IncludeAllContentForSelfExtract to false.</value>
```
Comment (agocke) [inline src/Tasks/Common/Resources/Strings.resx:661]: Or unset `IncludeNativeLibrariesForSelfExtract`, right? Maybe just state the contract, something like, "if IncludeAllContentForSelfExtract is true, IncludeNativeLibrariesForSelfExtract must not be false"?

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets [hunk 2]
```diff
@@ -94,6 +94,10 @@ Copyright (c) .NET Foundation. All rights reserved.
                  ResourceName="CannotHaveSingleFileWithoutExecutable" />
     <NETSdkError Condition="'$(PublishSingleFile)' == 'true' And '$(_IsExecutable)' == 'true' And '$(TargetFrameworkIdentifier)' != '.NETCoreApp'"
                  ResourceName="CanOnlyHaveSingleFileWithNetCoreApp" />
+    <NETSdkError Condition="'$(PublishSingleFile)' == 'true' And
+                            '$(IncludeSymbolsInSingleFile)' == 'true' And
+                            '$(_TargetFrameworkVersionWithoutV)' >= '5.0' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp'"
```
Comment (agocke) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets:99]: Does the `>` need to be escaped here (because XML)?

Comment (sbomer) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets:99]: `<` does, but `>` apparently doesn't - we use this pattern in a lot of places for version checks.

File: src/Tasks/Common/Resources/Strings.resx (hunk 1 above)
Comment (sbomer) [reply src/Tasks/Common/Resources/Strings.resx]: Fixed!

**Review by VSadov — APPROVED** [review]
LGTM

**Review by agocke — APPROVED** [review]
LGTM, thanks!

**Comment by agocke** [comment]
@sbomer any reason not to merge?

**Comment by sbomer** [comment]
You commented just as I was about to! :D

**Pull request merged**